### DISCUSSION TITLE
how to set the default toolbar ?

### DISCUSSION THREAD

**Discussion by icoco** [eclipse-theia/theia#13863] [Q&A] [answered]
from the online document: https://theia-ide.org/docs/toolbar/,  it guide implementation "ToolbarDefaultsFactory", but looks the "ToolbarDefaultsFactory" is  interface Symbol , not sure the details usage,  I assume that maybe have more simply way to customize the default 'ToolbarDefaults' eg: 
```
export const FolderToolbarDefaults: () => DeflatedToolbarTree = () => ({
    items: {
        [ToolbarAlignment.LEFT]:[],
        [ToolbarAlignment.CENTER]: [[
            {
                id:  FolderCommands.OPEN_FOLDER.id,
                command: FolderCommands.OPEN_FOLDER.id,
                icon: FolderCommands.OPEN_FOLDER.iconClass,
            },
        ]],
        [ToolbarAlignment.RIGHT]:[]
    } 
    
}); 

```
also bind it 
```
    if (isBound(ToolbarDefaultsFactory)) {
        rebind(ToolbarDefaultsFactory).toConstantValue(FolderToolbarDefaults);
      } else {
        bind(ToolbarDefaultsFactory).toConstantValue(FolderToolbarDefaults);
    }

```
but It does not effect at all, again, I can use ToolbarController addItem through menu command, it works fine ( I meaning add new toolbar item on the toolbar, but it does not work in the method 'initializeLayout' of the instance class 'FrontendApplicationContribution' .
 also not works in the callback event 'onWorkspaceChanged' of the class 'workspaceService'

 I am not sure whether the online guide is latest or anyone give best solution that can easy customize the default toolbar ?  thanks !

**Comment by msujew** [comment, marked as answer]
Hey @icoco,

in order for the toolbar defaults to take effect, you need to run the `Reset Workbench Layout` command. This will reset your layout to the default (including the toolbar). Otherwise, the toolbar will always be loaded from your browser's local storage.

After running the command, it should reset it to your specified default toolbar items.

**Reply by icoco** [reply]
thanks for guide, that would be great.

 I also find other way, remove the  folder :  '~.theia'  ( the toolbar.json store in the configuration path),  after that, it works
